tests/src/JIT/SIMD/Vector3Interop.cs

```diff
@@ -0,0 +1,339 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+
+public struct DT 
+{
+    public Vector3 a;
+    public Vector3 b;
+};
+
+[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+public struct ComplexDT 
+{
+    public int iv;
+    public DT vecs;
+    [MarshalAs(UnmanagedType.ByValTStr, SizeConst=256)] public string str;
+    public Vector3 v3;
+};
+
+class PInvokeTest 
+{
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern int nativeCall_PInvoke_CheckVector3Size();
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern float nativeCall_PInvoke_Vector3Arg(
+        int i, 
+        Vector3 v1, 
+        [MarshalAs(UnmanagedType.LPStr)] string s, 
+        Vector3 v2);
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern Vector3 nativeCall_PInvoke_Vector3Ret();
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern float nativeCall_PInvoke_Vector3Array(Vector3[] v_array);
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern DT nativeCall_PInvoke_Vector3InStruct(DT d);
+
+    [DllImport(@"Vector3TestNative", CallingConvention = CallingConvention.StdCall)]
+    public static extern void nativeCall_PInvoke_Vector3InComplexStruct(ref ComplexDT cdt);
+    
+    public static bool test() 
+    {
+
+        // Expected return value is 12 bytes.
+        if (nativeCall_PInvoke_CheckVector3Size() != 12) 
```

Comment:
The above is on Windows and a reference to copy of Vector3 is passed by native caller.  Managed callee would have to indirect on addr received as arg and RyuJIT would read just 12 bytes.

What about on x64 Unix?  Vector3 could be passed either in two registers or on stack.  If passed in registers, RyuJIT prolog homes Vector3 arg to a stack location. As per my understanding RyuJIT is homing 16 bytes and assumes upper 4 bytes are zero initialized.  Similarly, what happens when Vector3 arg is passed on stack, does RyuJIT generated code try to read 12 or 16 bytes?